### PR TITLE
[CI] catch a broken rev/non-rev config

### PR DESCRIPTION
**Pull request by blefaudeux**
## What does this PR do?
Catch a possible broken config, raised in https://github.com/facebookresearch/xformers/pull/45, where some layers would be reversible but not all

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [x] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 [![CircleCI](https://circleci.com/gh/facebookresearch/xformers.svg?style=shield)](https://app.circleci.com/pipelines/github/facebookresearch/xformers/)
 [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](CONTRIBUTING.md)
 [![codecov](https://codecov.io/gh/facebookresearch/xformers/branch/main/graph/badge.svg?token=PKGKDR4JQM)](https://codecov.io/gh/facebookresearch/xformers)
+[![black]("https://img.shields.io/badge/code%20style-black-000000.svg")]("https://github.com/psf/black">
```
Comment (blefaudeux) [inline README.md:9]: opportunistic, just add a [code: black] badge to warn about the formatting

File: xformers/factory/block_factory.py [hunk 2]
```diff
@@ -219,7 +219,7 @@ def __init__(
             feedforward_config=feedforward_config,
             position_encoding_config=position_encoding_config,
             layer_norm_style=LayerNormStyle(layer_norm_style),
-            block_type=BlockType("encoder"),
+            block_type=BlockType("decoder"),
```
Comment (dianaml0) [inline xformers/factory/block_factory.py:222]: Wow, nice catch

File: xformers/factory/model_factory.py [hunk 3]
```diff
@@ -55,10 +55,33 @@ def __init__(self, stack_configs: List[Dict[str, Any]]):
...
+        except AssertionError:
+            print(self.stack_configs)
+            print(reversible)
+            assert False
+
 
 class xFormer(torch.nn.Module):
     def __init__(
-        self, stack_configs: List[Union[xFormerStackConfig, xFormerStackConfig]]
+        self, stack_configs: Union[xFormerStackConfig, List[xFormerStackConfig]]
```
Comment (dianaml0) [inline xformers/factory/model_factory.py:84]: Much more convenient!

**Review by dianaml0 — APPROVED** [review]
Nice improvement to the code!

File: xformers/factory/block_factory.py (hunk 2 above)
Comment (blefaudeux) [reply xformers/factory/block_factory.py:222]: yep, this was not too good, my bad..

**Pull request merged**